### DISCUSSION TITLE
COSS 课程讨论汇总贴

### DISCUSSION THREAD

**Discussion by birdflyi** [X-lab2017/open-wonderland#163] [3️⃣ 聊学术]
可以在此“Discussion”下讨论与COSS课程相关的问题，具体日程可以[参见这里](https://github.com/X-lab2017/open-wonderland/blob/main/%F0%9F%8C%B32022-XSOSI-OpenCourse.md#%E7%AC%AC%E4%B8%89%E5%91%A8coss-%E8%AF%BE%E7%A8%8B)~

**Comment by bifenglin** [comment]
8 月 15日 的第二节课“Software Products”，欢迎大家在此条目下进行讨论~

**Comment by tyn1998** [comment]
# 关于梅赛德斯·奔驰给出的开源许可证信息的若干个问题

Dirk教授在COSS C01中给出一张关于奔驰的slide，并夸赞了奔驰的做法。

<img width="500" alt="image" src="https://user-images.githubusercontent.com/32434520/185017745-bebecf17-37c4-4387-b091-10d1e7ec855d.png">

奔驰不同车型用到的开源软件不完全相同，因此每个车型附带的这个记录许可证信息的光碟的内容也不相同。我在网上找到了[C级车的版本](https://moba.i.mercedes-benz.com/bai-cars/ba/foss/content/en/pdfs/FOSS_License_C-Class_205.pdf)。浏览一番后，我有如下几个问题，主要是关于“Copyright Notice”的。

## Q1：为什么看上去“差不多”的Copyright Notice要列这么多个？

奔驰的这个册子是按照开源许可证来组织的，好像是把所有用到某种开源许可证的开源软件的Copyright Notice都放到了一起。以“Acme Labs BSD License”为例，差不多都是Jef Poskanzer一个人的版权信息，只是年份不同，有些年份在我看来还是包含关系，感觉像是冗余重复的。为什么要这么做呢？

<img width="1368" alt="image" src="https://user-images.githubusercontent.com/32434520/185018768-a88a483e-e4e5-4682-9221-e1b2924b3283.png">

## Q2：MIT License有多个版本？

MIT许可证是常用的许可证，于是我浏览了相关的部分，发现似乎存在着好多个MIT许可证的版本，主要是第三段话有略微区别。每个版本最前面都是相关的Copyright Notice，第一个版本（和[官网](https://opensource.org/licenses/MIT)一致的版本）拥有的Copyright Notice最多。

<img width="1360" alt="image" src="https://user-images.githubusercontent.com/32434520/185019641-7db2898f-abb0-4105-bbc8-21b3d70a8f3c.png">

## Q3：如何更新Copyright Notice？

日常生活中总能看到©版权符号以及相应的Copyright Notice。在浏览了这个册子后，我了解到版权信息中的year很重要。那么：

- 什么时机要更新这个year呢？
- 是每年都去更新一下，还是有改动的时候才更新？
- 如果忘记更新了会怎么样？

## Q4：Copyright Notice一定要显式声明吗？

Dirk教授在举奔驰的例子时，强调了一下GPL2.0和下一个证书之间有一个“big jump”：

![image](https://user-images.githubusercontent.com/32434520/185021297-74c2d945-d99e-4b7c-89ab-63502df5f137.png)

他说这很可能是因为Linux Kernel项目引入的。看了一下Linux Kernel项目，大部分文件的开头都是有版权声明的，所以我估计奔驰是自动生成了这些版权信息。

<img width="1278" alt="image" src="https://user-images.githubusercontent.com/32434520/185022585-c398718e-a8f5-4d74-8213-b0068826a5bf.png">

我的问题是：没有显式声明版权信息的贡献者会自动得到版权吗？（不知道这样问对不对）

**Comment by tyn1998** [comment]
# 这张图看不懂。。

COSS C01第31张slide：

<img width="696" alt="image" src="https://user-images.githubusercontent.com/32434520/185027634-996f526d-8868-4a73-9c91-9c472995e415.png">

**Comment by tyn1998** [comment]
# 为什么apache/dubbo只要求有较多contribution的贡献者签署CLA？

当一个贡献者签署了某个开源项目/基金会的CLA，那么TA的代码的Copyright就转移到该开源项目/基金会了。CLA存在的主要目的是服务于vendor的，这样可以减少版权纠纷等一系列法律问题。

问题：Why [apache/dubbo](https://dubbo.apache.org/en/docs/contribution-guidelines/contributor/cla-signing-guide_dev/) only requires those who contribute a lot to sign CLA? How about occasional contributors?

<img width="1370" alt="image" src="https://user-images.githubusercontent.com/32434520/185030177-30539f8e-13d6-4755-be91-ce5bc93a4a05.png">

**Reply by zhicheng-ning** [reply]
版权许可能避免撤销贡献者的代码，专利许可能避免贡献者起诉它们专利侵权，并且，条款中还有贡献者原创声明——如果合并后的代码侵犯他人版权，那么这是贡献者的责任。至于为什么只要求有较多 contribution 的贡献者签署 CLA，我认为一个可能的原因是为了图方便，因为如果只要保证签署了 CLA 的贡献者所提交的代码占据代码总量的绝大部分 ，那么其余少数的贡献其实可以很容易进行简单的重构，所以相比于让所有贡献者签署 CLA（可能只是修了一个 typo），只要求有较多 contribution 的贡献者签署 CLA 是一个更高效率的方式。

**Comment by tyn1998** [comment]
# 另一张看不懂的图

COSS C02 42页：

<img width="691" alt="image" src="https://user-images.githubusercontent.com/32434520/185030441-79c9f9ef-0048-4928-afe2-425c101f7f92.png">

**Comment by zhicheng-ning** [comment]
关于 Commerical Open Source  和 Cloud Service Strategies 相关的内容可在此讨论

**Comment by yoyo-wu98** [comment]
关于 Software Vendor 和 Business Model 相关的内容可在此讨论
